syntax = "proto2";

package fasterrcnn.protos;

import "fasterrcnn/protos/eval.proto";
import "fasterrcnn/protos/graph_rewriter.proto";
import "fasterrcnn/protos/input_reader.proto";
import "fasterrcnn/protos/model.proto";
import "fasterrcnn/protos/train.proto";

// Convenience message for configuring a training and eval pipeline. Allows all
// of the pipeline parameters to be configured from one file.
// Next id: 8
message TrainEvalPipelineConfig {
  optional DetectionModel model = 1;
  optional TrainConfig train_config = 2;
  optional InputReader train_input_reader = 3;
  optional EvalConfig eval_config = 4;
  repeated InputReader eval_input_reader = 5;
  optional GraphRewriter graph_rewriter = 6;
  extensions 1000 to max;
}
